---
title: "DESeq2"
author: "Samuel Gurr"
date: "2023-02-13"
output: pdf_document
---


## Setup: 

```{r setup, include=TRUE}
knitr::opts_chunk$set(echo = TRUE, 
                      warning = FALSE, 
                      message = FALSE, 
                      cache = TRUE)

knitr::opts_knit$set(root.dir = "~/Documents/Github_repositories/Airradians_multigen_OA/RAnalysis")


```

#### Load libraries

```{r load_libraries, include = TRUE}
# load libraries - notes show the install command needed to install (pre installed)
library(DESeq2) # note: this was previously installed with the command `BiocManager::install("DESeq2")`
library(edgeR)
library(goseq)
library(dplyr)
library(GenomicFeatures)
library(data.table)
library(calibrate)
library(affycoretools) # note: this was previously installed with the BiocManager::install("affycoretools")
library(data.table)
library(vsn)
library(tidybulk)
# Plotting
library(ggplot2)
library(cowplot)
library(pheatmap)
library(gplots)
library(RColorBrewer)
library(EnhancedVolcano)  # note: this was previously installed with the command `BiocManager::install("EnhancedVolcano")`
library(pcaExplorer) # note: this was previously installed with the command `BiocManager::install("pcaExplorer")
```

#### Set path

```{r set the path out, include = TRUE}
path_out = 'C:/Users/samjg/Documents/Github_repositories/Airradians_multigen_OA/RAnalysis/Output/Transcriptomics/DESeq2'
```





# LOAD DATA: choose ONE of the counts clusters (filtered raw reads or unfiltered raw reads)



### Gene count data - convert to matrix - NOTE ONLY RUN ONE (FILTERED  versus UNFILTERED)
I include both filtered and unfiltered here to compare the results. 
DESEq2 mentions that pre-filtering is not necessary however will reduce the memory/time to run dds and rlog of the models 
Below are both RAW counts filtered and unfiltered to run DESeq2 

# FILTERED COUNTS:  3 CPM (with edgeR) and 50% of samples 
```{r call the count matrices, include = TRUE}
getwd()
### raw counts tables 
raw_counts_matrix_annotation  <- 
  read.csv(file="../../Output/Transcriptomics/Raw_counts_matrix/raw_count_matrix_WITH_ANNOTATION.csv",
                                               sep=',', 
                                               header=TRUE) %>% dplyr::select(!c(Ai19, Ai3))
counts_matrix <- 
  data.frame(raw_counts_matrix_annotation[,-c(1,41:44)], #[,-c(1,39:42)] - if you run dplyr::select(!c(Ai19, Ai3)) above
             row.names=raw_counts_matrix_annotation[,1])
### filtered counts tables 3CPM - format matrix after upload [from Count_Matrix_Stats.Filter.R] 
filtered_counts_matrix_annotation  <-
  read.csv(file="../../Output/Transcriptomics/Filtered_counts_matrix/filter_3cmpm50perc_WITH_ANNOTATION.csv", 
                                               sep=',', 
                                               header=TRUE)  %>% dplyr::select(!c(Ai19, Ai3))
counts_matrix <- 
  data.frame(filtered_counts_matrix_annotation[,-c(1,41:44)], # [,-c(1,39:42)] - if you run dplyr::select(!c(Ai19, Ai3)) above
             row.names=filtered_counts_matrix_annotation[,1]) 


ncol(counts_matrix) # 39 samples
```



### Sample metadata - Experimental treatments/groups
```{r experiment_data, include = TRUE}
### experiment metadata [from Count_Matrix_Stats.Filter.R]  - convert characaters to factors for DESeq2
exp_data  <- read.csv(file="../../Data/Transcriptomics/exp.metadata.csv", sep=',', header=TRUE)
```



### DEsign DESeqDatasets
```{r build_dds_datasets}
# ========================================================== 
#  FULL MODEL ==  design = ~ pCO2
# ========================================================== #
#####
Exp.metadata       <- exp_data %>% 
                      dplyr::select(c('SampleName_readmatrix', 'pCO2')) %>% 
                      dplyr::rename(Sample.Name = SampleName_readmatrix) %>% # coondense dataset to build target matrix
                      dplyr::filter(Sample.Name %in% colnames(counts_matrix))
Exp.metadata.df    <- data.frame(Exp.metadata[,-1], row.names=Exp.metadata[,1]) # move Sample.Name column as row names  
Exp.metadata.mtx   <- as.matrix(Exp.metadata.df, row.names="BayScallop.ID") # create matrix 

# check for 'TRUE' in each - check before proceeding  design
Exp.metadata.mtx_2 <- as.data.frame(Exp.metadata.mtx[match(colnames(counts_matrix),rownames(Exp.metadata.mtx)), ]) %>% rename(pCO2 = 1)
all(rownames(Exp.metadata.mtx_2) %in% colnames(counts_matrix)) # should be TRUE
all(rownames(Exp.metadata.mtx_2) == colnames(counts_matrix)) # should be TRUE


# build dds
dds <- DESeqDataSetFromMatrix(countData = counts_matrix,
                                      colData = Exp.metadata.mtx_2,
                                      design = ~pCO2) # DESeq Data Set (dds) - design as ~Primary_Treatment

```


## Transform Exp data and run diagnostic plots

* Why?
 - the transformed data can be used to run simple prince component analysis to view how samples (by treatment) relate
 - identify whether samples appear as outliers sufficient for omission to rerun dds.run

```{r expression data transformation}
# Data transformations for heatmap and PCA visuals ======================================================================================= #

# vstExp  :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

vstExp  <- vst(dds)

# histogram and meadSD plot ------------ #

png("../../Output/Transcriptomics/DESeq2/vstExp_histogram.png", 1000, 1000, pointsize=20)# diagnostics of transformation 
hist(assay(vstExp)) # view histogram 
dev.off() # write
png("../../Output/Transcriptomics/DESeq2/vstExp_mean_sd.png", 1000, 1000, pointsize=20)
meanSdPlot(assay(vstExp)) # shows the sd y axis (sq root of varaince in all samples) - flat curve may seem like a goals, BUT may be unreasonable in cases with MANY true DEGs from experimental conditions
dev.off() # write

# PCA plot vst  ------------------------ #

pcaData_vstExp    <- plotPCA(vstExp, intgroup = "pCO2", returnData = TRUE)
percentVar_vstExp <- round(100 * attr(pcaData_vstExp, "percentVar"))
png("../../Output/Transcriptomics/DESeq2/vstExp_PCA.png", 1000, 1000, pointsize=20)
ggplot(pcaData_vstExp, aes(x = PC1, y = PC2, color = pCO2, label=name)) +
  scale_shape_manual(values = c(4, 19, 17)) +
  geom_text(aes(label=name),hjust=0.2, vjust=1.4, size=5) +
  geom_point(size =6) +
  theme_classic() +
  theme(text = element_text(size=15)) +
  ggtitle("PCA: vst expression data") +
  xlab(paste0("PC1: ", percentVar_vstExp[1], "% variance")) +
  ylab(paste0("PC2: ", percentVar_vstExp[2], "% variance")) +
  coord_fixed()
dev.off()



# rlogExp ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

rlogExp <- rlogTransformation(dds) 
# rlog - regularized log transformation of origin count data to log2 scale 
# fit for each sample and dist. of coefficients in the data

# histogram and meadSD plot ------------ #

png("../../Output/Transcriptomics/DESeq2/rlogExp_histogram.png", 1000, 1000, pointsize=20)# diagnostics of transformation 
hist(assay(rlogExp)) # view histogram 
dev.off() # write
png("../../Output/Transcriptomics/DESeq2/rlogExp_mean_sd.png", 1000, 1000, pointsize=20)
meanSdPlot(assay(rlogExp)) # shows the sd y axis (sq root of varaince in all samples) - flat curve may seem like a goals, BUT may be unreasonable in cases with MANY true DEGs from experimental conditions
dev.off() # write

# PCA plot rlog  ----------------------- #

pcaData_rlogExp    <- plotPCA(rlogExp, intgroup = "pCO2", returnData = TRUE)
percentVar_rlogExp <- round(100 * attr(pcaData_rlogExp, "percentVar"))
png("../../Output/Transcriptomics/DESeq2/rlogExp_PCA.png", 1000, 1000, pointsize=20)
ggplot(pcaData_rlogExp, aes(x = PC1, y = PC2, color = pCO2, label=name)) +
  scale_shape_manual(values = c(4, 19, 17)) +
  geom_text(aes(label=name),hjust=0.2, vjust=1.4, size=5) +
  geom_point(size =6) +
  theme_classic() +
  theme(text = element_text(size=15)) +
  ggtitle("PCA: rlog expression data") +
  xlab(paste0("PC1: ", percentVar_rlogExp[1], "% variance")) +
  ylab(paste0("PC2: ", percentVar_rlogExp[2], "% variance")) +
  coord_fixed()
dev.off()

```
### Plot reads per gene and reads per sample in each dds

```{r ddsPlots_read.gene_reads.sample}
# DAY 0 PLOT THE (1) READS PER SAMPLE (2) READS PER GENE - FOR EACH DDS
nsamples        <- ncol(counts(dds)) # Number of samples - for the plot label
rps             <- qplot(colSums(counts(dds))) +# reads of reads per sample 
                    labs(x = "Mapped reads per sample", y = "Number of samples",
                         title = "Day 0: Mapped reads per sample") +
                    geom_label(aes(x = 1.5e6, y = 3, label = paste(nsamples, "samples")))
ngenes          <- nrow(counts(dds)) # Number of genes
ngenes_min      <- min(rowSums(counts(dds))) #  minimum reads
ngenes_median   <- median(rowMeans2(counts(dds))) # minimum row mean
ngenes_max      <- max(rowSums(counts(dds))) #  maximum reads
rpg             <- qplot(log10(rowSums(counts(dds))), bins = 16) + # number of reads per gene
                    labs(x = "log10(Mapped reads per gene)", 
                         y = "Number of genes",
                         title = "Mapped reads per gene") +
                    geom_label(aes(x = 4, y = 2000, label = paste(ngenes, "total genes"))) +
                    geom_label(aes(x = 4, y = 1700, label = paste("max count =", ngenes_max))) +
                    geom_label(aes(x = 4, y = 1400, label = paste("min count =", ngenes_min))) +
                    geom_label(aes(x = 4, y = 1100, label = paste("median reads =", ngenes_mean.min)))
countfig <- plot_grid(rps, rpg)
countfig


```


### RUN DESEQ2 model - wait for this to complete...

# NOTE: this will be longer dependent on filtered vs. unfiltered raw read count data for dds objects 
```{r run_dds}
dds.run <- DESeq(dds) # wait for this to complete....
```


## Differential expression

### Main Treatment effects: padj < 0.05; FDR 5%; LFC >1 (<-1)

* pCO2: Low v. High

```{r Examine DEGs how many? what are they?}

# Exxplore the 'dds.run' model
resprimary         <- results(dds.run, contrast=c("pCO2", "High", "Low"), alpha = 0.05) # FDR is the alpha 0.05
numDEGs_padj       <- data.frame(table(resprimary$padj<0.05))[2,2] # DEGs - NOT considering LFC - just p adj
numDEGs_pval       <- data.frame(table(resprimary$pvalue<0.05))[2,2] # DEGs - NOT considering LFC - just p adj
resprimary.ordered <- resprimary[order(resprimary$padj), ] # Order by adjusted p-value
num.UpReg          <- sum((resprimary.ordered$log2FoldChange[1:numDEGs_padj] > 0) == TRUE) #  LFC >= 1
num.DownReg        <- sum((resprimary.ordered$log2FoldChange[1:numDEGs_padj] < 0) == TRUE) # LFC >= 1
total              <- sum(num.DownReg,num.UpReg) # sum of DEGs with the criteria pdj < 0.05 + LFC>1 (< -1)
# Write results - covert to as.data.frame for the ordered results
resdata.primary           <- merge(as.data.frame(resprimary.ordered), 
                                    as.data.frame(counts(dds.run, normalized=TRUE)), 
                                    by="row.names", 
                                    sort=FALSE) ## Merge with normalized count data
names(resdata.primary)[1] <- "Gene"
resdata.primary           <- resdata.primary[order(resdata.primary$padj), ] # Order by adjusted p-value
write.csv(resdata.primary, "../../Output/Transcriptomics/DESeq2/DESeq2results.csv") # write
# What are the Protein names of these DEGs?
# first gather gene IDs for the DEGs (note: this is also used for the pheatmap in the transformed Exp heatmap)
numDEGs_padj # 54 - number of DEGs (from above) - use as integer in call below
topgenes.IDs <- head(rownames(resprimary.ordered),numDEGs_padj) # call gene IDs for DEGs: pdj < 0.05 FDR < 0.05 
length(topgenes.IDs) == numDEGs_padj # must be TRUE
# now load in the annotated csv
Airr_Cvirg_annotation <- read.csv(file="../../Output/Transcriptomics/Raw_counts_matrix/raw_count_matrix_WITH_ANNOTATION.csv", 
                                  sep = ',', 
                                  header = T) %>% 
                        dplyr::select(c('Airradians_TranscriptID',
                                 "blastxEval_CvirgTranscriptID",
                                 "blastxEval_CvirgProteinID",
                                 "blastxEval_CvirgGeneID",
                                 "blastxEval_CvirgGOterms"))
# DEGs Protein IDs - lets take a look shall we! 
DEGs_annotated <- Airr_Cvirg_annotation %>% filter(Airradians_TranscriptID %in% topgenes.IDs)
nrow(DEGs_annotated) # 52 - all are present! 
View(DEGs_annotated)
```

## Plot the DEG statistics

```{r histogram}
# histogram of FDR
png("../../Output/Transcriptomics/DESeq2/FDR_hist.png", 1000, 1000, pointsize=20)
hist(resprimary$padj, breaks=20, col="grey") # view histogram
abline(h=c( (nrow(resprimary)*0.05), 
            ((table(resprimary$padj < 0.1)[2]) + (nrow(resprimary)*0.1)),
            ((table(resprimary$padj < 0.05)[2]) + (nrow(resprimary)*0.05)) ),
                  col=c("blue", "red", "red"), lty=c(1,2,1), lwd=c(1, 3, 1)) # add line at 
dev.off()

# volcano plot 
png("../../Output/Transcriptomics/DESeq2/PrimaryTreatment-VolcanoPlot.png", 1000, 1000, pointsize=20)
EnhancedVolcano(resprimary,
                lab = rownames(resprimary),
                x = 'log2FoldChange',
                y = 'padj',
                title = 'pCO2 Treatment (Low v High)',
                subtitle = "DESeq2 - Differential expression",
                FCcutoff = 1,
                pCutoff = 0.05,
                pointSize = 4.0,
                labSize = 6.0,
                colAlpha = 1,
                legendPosition = 'right',
                legendLabSize = 12,
                legendIconSize = 4.0,
                widthConnectors = 0.75)
dev.off()


# Plot dispersion 

png("../../Output/Transcriptomics/DESeq2/dispersions.png", 1000, 1000, pointsize=20)
plotDispEsts(dds.run, main="dispersions")
dev.off()

```


* plot heatmap

```{r heatmap for transformed data}

# Plot pheatmap map rlog
# setup the save pheatmap function
save_pheatmap <- function(x, filename, width=1000, height=960) { # template for saving pheatmap outputs
  stopifnot(!missing(x))
  stopifnot(!missing(filename))
  png(filename,width = width, height=height)
  grid::grid.newpage()
  grid::grid.draw(x$gtable)
  dev.off()
}
# colors for the heatmap
# colorblind palette
colorBlindBlack8  <- c("#000000", "#E69F00", "#56B4E9", "#009E73", 
                         "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
pie(rep(1, 8), col = colorBlindBlack8) # view the pie chart for colorblindness
ann_colors = list(
    pCO2 = c(Low="#56B4E9", High="#E69F00"))
df_annot.col         <- as.data.frame(colData(dds.run)[c("pCO2")])


# rlogExp
# use topgenes.IDs from the chunk above - this calls only the pdj > 0.05 from the output results of 'dds.run'
topgenes_rlog.counts <- assay(rlogExp)[topgenes.IDs,] 
topgenes_rlog.corrected   <- topgenes_rlog.counts - rowMeans(topgenes_rlog.counts) # subtract from the row mean to get +/- 0 to normalize and ease the aesthetic
df_annot.col         <- as.data.frame(colData(dds.run)[c("pCO2")])
rlog.heatmap <- pheatmap(topgenes_rlog.corrected, 
                            annotation_col=df_annot.col, 
                            main = "Ambient versus Moderate (52 total DEGs)",
                            cutree_cols = 7,
                            cutree_rows = 2,
                            annotation_legend = TRUE,
                            annotation_colors = ann_colors,
                            show_rownames = TRUE, # Boolean for gene names
                            labels_col=df_annot.col$pCO2, 
                            angle_col = 0,
                            fontsize = 8,
                            legend = TRUE)
save_pheatmap(rlog.heatmap, filename = "../../Output/Transcriptomics/DESeq2/rlog_heatmap.png") #Save pheatmap



# vstExp
# use topgenes.IDs from the chunk above - this calls only the pdj > 0.05 from the output results of 'dds.run'
topgenes_vst.counts      <- assay(vstExp)[topgenes.IDs,] 
topgenes_vst.corrected   <- topgenes_vst.counts - rowMeans(topgenes_vst.counts) # subtract from the row mean to get +/- 0 to normalize and ease the aesthetic
vst.heatmap <- pheatmap(topgenes_vst.corrected, 
                            annotation_col=df_annot.col, 
                            main = "Ambient versus Moderate (52 total DEGs)",
                            cutree_cols = 2,
                            cutree_rows = 2,
                            annotation_legend = TRUE,
                            annotation_colors = ann_colors,
                            show_rownames = TRUE, # Boolean for gene names
                            labels_col=df_annot.col$pCO2, 
                            angle_col = 0,
                            fontsize = 8,
                            legend = TRUE)
save_pheatmap(vst.heatmap, filename = "../../Output/Transcriptomics/DESeq2/vst_heatmap.png") #Save pheatmap

```

```{r plot the rlogExp}

rlogExp_df        <- as.data.frame(assay(rlogExp))

rlogExp_df        <- tibble::rownames_to_column(rlogExp_df, "Gene.ID")

rlogExp_DEGs      <- rlogExp_df %>% 
                        dplyr::filter(Gene.ID %in% topgenes.IDs)

rlogExp_DEGs_melt <- reshape2::melt(rlogExp_DEGs, id.vars='Gene.ID') %>% 
                        dplyr::rename(Sample.Name = variable, # to merge with 'Exp.metadata'
                                      rlogExp = value) # the data to get mean SE values 

DEGs_annotated   <- Airr_Cvirg_annotation %>% 
                        dplyr::filter(Airradians_TranscriptID %in% topgenes.IDs) %>% 
                        dplyr::select(Airradians_TranscriptID,blastxEval_CvirgProteinID) %>% 
                        dplyr::rename(Gene.ID = Airradians_TranscriptID,
                                      Protein.ID = blastxEval_CvirgProteinID)

rlogExp_DEGs_annotated <- merge(DEGs_annotated, rlogExp_DEGs_melt, by = "Gene.ID")

rlogExp_DEGs_MASTER    <- merge(rlogExp_DEGs_annotated,Exp.metadata, by = "Sample.Name")

rlogExp_DEGs_MEAN.SE   <- rlogExp_DEGs_MASTER %>% 
                          dplyr:::group_by(Gene.ID, Protein.ID,pCO2) %>% 
                          dplyr::summarise(mean.rlogExp    = mean(rlogExp),
                                           sd.rlogExp     = sd(rlogExp),
                                           n = n(),
                                           sderror.rlogExp = sd.rlogExp/(sqrt(n))) %>% 
                          dplyr::select(-n) %>% 
                          dplyr::mutate(Protein_contig = paste(substr(Protein.ID,1,30),
                                                               str_split(Gene.ID, "\\.", simplify=T)[,3],
                                                               sep="_"))
pd <- position_dodge(0.3)



for(i in 1:length(topgenes.IDs)) {
  
  geneID_loop  <- topgenes.IDs[i]
  
  df_loop      <- rlogExp_DEGs_MEAN.SE %>% 
                  dplyr::filter(Gene.ID %in% geneID_loop) %>% 
                  dplyr::mutate(Protein_contig = str_remove_all(Protein_contig,"/"))
  
  plot_loop    <- df_loop %>% 
                    ggplot(aes(x=pCO2, 
                               y=mean.rlogExp, 
                               fill=pCO2)) +  # , colour=supp, group=supp))
                    theme_classic() +
                    geom_errorbar(aes(ymin=mean.rlogExp-sderror.rlogExp, 
                                      ymax=mean.rlogExp+sderror.rlogExp), 
                                  colour="black", 
                                  width=.1,
                                  position=pd) +
                    geom_point(size = 4, 
                               shape=21,
                               position=pd) +            
                    xlab("pCO2 treatment") +
                    ylab("") +                 # note the mean was first by sample ID THEN by treatment
                    scale_fill_manual(values=c("white",
                                               "grey50")) +
                    ggtitle(df_loop[1,7]) +
                    theme(axis.text.x = element_text(size = 20),
                          axis.text.y = element_text(size = 20),
                          axis.ticks.length=unit(.25, "cm")) +
                    theme(legend.position = "none")
  
if(df_loop[1,4]>df_loop[2,4]) { # Upregulated, high pCO2 == high expression
  pdf(paste("../../Output/Transcriptomics/DESeq2/Plots_MeanSE/Upregulated_DEGs/",df_loop[1,7],".pdf", sep = ''), width=5, height=5)
  print(plot_loop)
  dev.off() 
  } else {
  pdf(paste("../../Output/Transcriptomics/DESeq2/Plots_MeanSE/Downregulated_DEGs/",df_loop[1,7],".pdf", sep = ''), width=5, height=5)
  print(plot_loop)
  dev.off()  
  }


}
library(psych)

```

