---
title: "F1 Markdown Summary"
author: "Samuel Gurr"
date: "2023-01-14"
output: html_document
---

  * Last Updated: January, 2023

```{r load libraries, echo=FALSE, message=FALSE, warning=FALSE, include=FALSE}
# LOAD LIBRARIES 
library(ggplot2)
library(tidyverse)
library(dplyr)
library(tidyr)
library(tidyverse)
library(car)
library(nlme)
library(Rmisc)
library(lmerTest)
library(lme4)
library(pander)
library(performance)
library(Rmisc)
library(reshape2)
library(ggpubr)
```
# Table of Contents

----

| Headers | Subsections | Chunks & Figures |
|----|----|----|
|<a href="#Multigenerational Bay Scallop OA Exposure">Introduction</a><br>| <a href="#Big Picture Overview">Overview</a><br>  | [Figure 1](#Fig_1) |
|<a href="#F0 Broodstock">F0 Broodstock</a><br>| <a href="#About Broodstock">About broodstock</a><br> <a href="#Spawning and fertilization">Spawning and fertilization</a><br>   | <add here>       | 
|<a href="#F1: First Generation">First Generation</a><br>               | <a href="#F1 Larval development">Larval development</a><br> <a href="#Grow Out (F1s)">Grow out</a><br> <a href="#F1 Physiology">Physiology</a><br>  <a href="#Gene expression (F1s)">Gene expression</a><br>  <a href="#SNPs & Genotype liklihoods">SNPs & Genotype liklihoods (F1s)</a><br> | [Image: larvae treatments](#Img_1)    [Figure 2: Larvae survival](#Fig_2) | 
|<a href="#F2: First Generation">Second Generation</a><br>              | <a href="#Larval development (F2s)">Larval development</a><br> <a href="#Grow Out (F2s)">Grow out</a><br> <a href="#Physiology (F2s)">Physiology</a><br>  <a href="#Gene expression (F2s)">Gene expression</a><br>  <a href="#SNPs & Genotype liklihoods">SNPs & Genotype liklihoods (F2s)</a><br>| <add here>       | 
|<a href="#Multigenerational Bay Scallop OA Exposure">Introduction</a><br>| <add here>   | <add here>       | 

 

# Multigenerational Bay Scallop OA Exposure {#test}



We successfully created the F1 bay scallop cohort in late July 2021. These bay scallops were raised to maturity and spawned in August 2022 to create the F2 generation (Figure 1). Here we present a brief update of the data we have collected over the past year and where we are in our overall experimental plans. 

We have had many setbacks tin 2022 (Pseudomonas, pump failures, food issues, Vibrio), but after three spawnings, we were able to produce our F2 bay scallop generation. We collected a suite of physiology data during the F1 growout and Sam has made fantastic progress with library preparation for low coverage Whole Genome Sequencing and has sent his first test lane to be sequenced. 



Live links to summaries of each life state / generation within the document:

### Big Picture Overview

* Figure 1: Experimental design.

**F1 Larval development**

* Figure 2: Larval pictures at age 2 days
* Figure 3: F1 Larval Survival.
* Figure 4.1: F1 larval growth
* Table 1: Details of the water chemistry for each OA treatment. Means are in bold and standard error in parenthesis.

**F1 Grow Out**

* Figure 5: Experimental flow through set up
* Figure 6: Changes in temperature overtime for the F1 grow out.
* Figure 7: pCO2 over time during F1 growout
* Table 2: Average Water Chemistry during the F2 grow out

**F1 Physiology**

* Figure 8: F1 shell length spat - adult (high OA not included)
* Table 3: Log of physiology time points for F1 bay scallop grow out.
* Figure 9: Respiration rates (whole dry weight corrected).
* Figure 11: Biodeposition metrics for F1s. Updated 20221024.
* Figure 12: Gonad index over time.

**F2 Larval development**

* Table 4: Average Water Chemistry during the F2 larval experiment
* Table 5: Endpoint data summary for the F2 larval exposure to low and moderate OA
* Figure A: Larval survival during static exposures.
* Figure B: F2 Larvae mean shell length over time
* Metamorphosis
* Figure C: Percent successful metamorphosis (F2)

**F2 Grow Out**

* Figure D: pH over time during the change in pH from 8 to 7 in the low OA → high OA cohort/treatment.
* Figure E: pCO2 over time during the change in pH from 8 to 7 in the low OA → high OA cohort/treatment.
* Figure D: Mean shell growth rate (mm / month)
* Figure F: F2 juvenile growth: Mean shell length over time.
* Table X: Log of physiology time points for F2 bay scallop growout.

**Chemistry**

* Figure C1: Changes in water chemistry over time during F1 growout
* Figure C2: Changes in water chemistry over time during F1 growout






## Big Picture Overview

----

<a id="Fig_1"></a>
```{r Figure 1, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="**Figure 1: Experimental design.** This figure has been adapted from our original proposal to reflect our current status and plans for FY23 and our final year of this project. Blue shading indicates stages that have been successfully completed and identifies where we are now in the three year plan. In the F2 growout, we used extra spat from the pH = 8 larval exposure to initiate a post settlement pH = 7 cohort. This is an alternative since we were not able to produce a pH=7 cohort that survived the larval exposure.", out.width = '80%'}
knitr::include_graphics("C:/Users/samjg/Documents/Github_repositories/Airradians_multigen_OA/Rmd_summary/images/Overivew.jpg")
```


## F0 Broodstock and spawning

---

#### About Broodstock

  In late July 2021 we spawned bay scallops produced from Massachusetts broodstock (F0). We had initially hoped for two populations, but the Connecticut population are now scarce and what few broodstock we did find, did not survive transport to the lab and the conditioning period. 


#### Spawning and fertilization

  Spawning was induced using thermal adjustments up to 25°C. When an individual was observed to be spawning, it was removed from the spawning table and placed in a 1-L beaker with filtered seawater at 25°C to continue releasing gametes. To avoid self fertilization, scallops were switched to new beakers frequently to isolate sperm and egg from each individual. When eggs were observed to be released, the animal was immediately switched to a new cup and eggs were caught on a 25µm sieve to remove any sperm that remained in the beaker. Prior to fertilization, sperm was rinsed through a 20µm sieve to remove any eggs that may have remained in the cup. Fertilizations were completed in batches to assure gametes were fertilized within approximately 1 hour of release and for each batch fertilization, each individual used contributed only sperm or only eggs to avoid self fertilization. Embryos were checked frequently to assure fertilization was occurring and gametes were not pooled until > 85% of the embryos showed clear cell division. Eggs were used from a total of 13 individuals and sperm from a total of 16 individuals, of these 6 individuals supplied both sperm and egg, but never to the same fertilization set. 

## F1: First Generation

---



#### Exposure to treatments

  After approximately 4 hours post fertilization, embryos from all batches were pooled and divided among 3 OA treatments (~400, 800, and 1600 µatm) with 5 replicates per treatment. Exposures were conducted in 10L filtered seawater at 27 psu and 23°C. Details of the chemistry for each OA treatment are detailed in Table 1. 

**Table 1: Seawater chemistry** Details for each OA treatment. Means are in bold and standard error in parenthesis.  

|Treatment|pCO2|pH|ΩAr|
|---|---|---|---|
|High OA |1788 | 7.41 | 0.6  |
|Mid OA  |829  | 7.72 | 1.21 |
|Low OA  |474  | 7.93 | 1.85 |

<a id="Fig_1_SW_chemistry"></a>
```{r Fig_1_SW_chemistry, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="**Figure Figure 1 seawater chemistry.** Average pCO2, temperature, Salinity, and aragonite saturation state", out.width = '100%'}

carbchem.MASTER <- read.csv(file="C:/Users/samjg/Documents/Github_repositories/Airradians_multigen_OA/RAnalysis/Output/SeawaterChemistry/F1s/master/CarbChem_master.csv", header = TRUE)


# change to long format to properly facet
carbchem.MASTER_long <- carbchem.MASTER %>% 
  dplyr::select(!c(Treatment,Replicate)) %>% 
  melt(id.vars=c('Date', 'Type', 'pCO2_treatment'))

carbchem.MASTER_long$value <- as.numeric(carbchem.MASTER_long$value) # make numeric

# aquire means by group for mean SE plotting by Date and treamtent
carbchem.MASTER_MEANS <- carbchem.MASTER_long %>% # calc means and standard error
   na.omit() %>% 
   dplyr::group_by(Date, Type, pCO2_treatment, variable) %>% 
   dplyr::summarise(mean = mean(value),
                    sd   = sd(value),
                    se   = sd/(sqrt(n())) )

# plot em!

pd <- position_dodge(0.1) # adjust the jitter for the different treatments   

All <- carbchem.MASTER_MEANS %>% 
  #dplyr::filter(Type %in% 'Adults') %>% 
  dplyr::filter(variable %in% (c('pCO2_out_matm',
                                 'Salinity',
                                 'Temperature_bucket_C',
                                 'pH_out',
                                 'War_out_DIC_pH'))) %>% 
  dplyr::mutate(Date = as.character(as.Date(Date, format = '%m/%d/%Y')))

#dates <- unique(sort(All$Date))
All$Date <- factor(All$Date, levels = unique(sort(All$Date)))

# day   <- sub('([^/])*.','',(sub('/([^/]*)$','',carbchem.MASTER_MEANS$Date)))
# month <- sub('/([^/].*)$','',carbchem.MASTER_MEANS$Date)
# year  <- sub('.*([/^])','',carbchem.MASTER_MEANS$Date)




plot <- ggplot(data= All, aes(x=Date, y=mean, shape=Type, colour=factor(pCO2_treatment))) + 
  geom_line(position=pd) +
  geom_errorbar(aes(ymin=mean-se, ymax=mean+se), colour="black", width=.1, position=pd) +
  scale_colour_manual(values = c("Low" = "green",
                                 "Moderate"="orange",
                                 "Severe" = "purple")) +
  theme_classic() +
  geom_point(position=pd, size=2) +
  #scale_x_discrete(limits=All$Date, breaks=All$Date[seq(1,length(All$Date), by = 50)]) +
  #theme(axis.text.x = element_text(angle = 45, hjust=1, size = 5)) +
  #theme(axis.text.x = NA) +
  #scale_y_continuous(expand = c(0, 0), limits = c(0, 300000)) +
  facet_wrap(~variable, scales = "free") 

print(plot)
```



### F1 Larval development



<a id="Img_1"></a>
```{r Image 1, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="**Figure 2: Larval pictures at age 2 days.** Larval pictures at age 2 days post fertilization showing distinct differences in shell development to D stage.", out.width = '100%'}
knitr::include_graphics("C:/Users/samjg/Documents/Github_repositories/Airradians_multigen_OA/Rmd_summary/images/F1_Larvae.jpg")
```



<a id="Fig_2"></a>
```{r Survival plot Mean SE, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="**Figure 2: F1 Larval Survival.** Survival over time. Error bars represent standard error. Settlement was evident by age 10 days, therefore survival counts are likely to be underestimations as a result of very sticky scallops making it difficult to suspend. (last updated 9/20/22)", out.width = '80%'}

# read csv file 
F1_survival <- read.csv(file= "C:/Users/samjg/Documents/Github_repositories/Airradians_multigen_OA/RAnalysis/Data/Physiology/Survival/F1/cumulative_raw/F1_survival_raw.csv", header = TRUE)
#First make calculations for means and standard error
LarvaeSurvival_Summ <- Rmisc::summarySE(F1_survival, 
                                 measurevar="Survival", 
                                 groupvars=c("Day", "Treatment")) %>% 
    dplyr::mutate(pCO2 = case_when(
      Treatment == "Low"  ~ "500 μatm", 
      Treatment == "Moderate" ~ "800 μatm", 
      Treatment == "High" ~ "1200 μatm")) # chamber volumes on different dates with different instruments 

LarvaeSurvival_Summ$pCO2 <- factor(LarvaeSurvival_Summ$pCO2, c("500 μatm","800 μatm","1200 μatm"))

## Use geom_line and geom_point to plot over time
LarvaeSurvival_Plot <-ggplot(data=LarvaeSurvival_Summ, 
                             aes(x=Day, y=Survival, 
                                 color=pCO2)) +
  geom_line( stat = "identity", size=1.0)+
  geom_point()+ 
  scale_color_manual(values=c("darkorange2","forestgreen", "purple"))+
  geom_errorbar(aes(ymin=Survival-se, ymax=Survival+se), width=.2,
                position=position_dodge(.1))+
  theme_classic() +  
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+ 
  scale_x_continuous(name ="Age (days)") +
  scale_y_continuous(name ="Survival (%)")+
  theme(text = element_text(size=15))
print(LarvaeSurvival_Plot)


```



<a id="Fig_3"></a>
```{r Larval length plot Mean SE, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="**Figure 3: F1 larval growth** Mean shell length overtime. Settlement was evident in the low and moderate OA treatments beginning at age 10 days.", out.width = '80%'}

# Early length  ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
# read csv file 
F1_length <- read.csv(file= "C:/Users/samjg/Documents/Github_repositories/Airradians_multigen_OA/RAnalysis/Data/Physiology/Length/F1/cumulative_raw/F1_lengths_raw.csv", header = TRUE)
# means for plotting
F1_length_MEANS         <- summarySE(F1_length, measurevar="Length_um", groupvars=c("Age_DPF", "pH", "Life_stage"))
F1_length_MEANS_by_tank <- (summarySE(F1_length, measurevar="Length_um", groupvars=c("Age_DPF", "pH", "Replicate")))

#plot
F1_length_MEANS <- F1_length_MEANS %>% 
  dplyr::mutate(pCO2 = case_when(
    pH == 8 ~ "500 μatm",  
    pH == 7.5 ~ "800 μatm", 
    pH == 7 ~ "1200 μatm"))
F1_length_MEANS$pCO2 <- factor(F1_length_MEANS$pCO2, c("500 μatm","800 μatm","1200 μatm"))

## Use geom_line and geom_point to plot over time
F1_length_lineplot_LARVAE <- F1_length_MEANS %>% 
                                    dplyr::filter(Life_stage %in% 'larvae') %>% 
                                    mutate(Life_stage = factor(Life_stage, 
                                                               levels = c("larvae", 
                                                                          "spat", 
                                                                          "juvenile", 
                                                                          "adult"))) %>% # relevel for the facets to occur in correct order
                                      ggplot(aes(x=Age_DPF, 
                                                 y=(Length_um/1000), 
                                                 color=pCO2)) +
                                        geom_line(size = 0.5) +  # aes(linetype = factor(pCO2)),
                                        geom_point()+ 
                                        scale_color_manual(values=c("forestgreen", "darkorange2","purple"))+
                                        geom_errorbar(aes(ymin=(Length_um/1000)-(se/1000),
                                                          ymax=(Length_um/1000)+(se/1000)), width=.2,
                                                      position=position_dodge(.1))+
                                        theme(panel.grid.major = element_blank(), 
                                              panel.grid.minor = element_blank())+ 
                                        scale_x_continuous(name ="Age (days)") +
                                        scale_y_continuous(name ="Shell length (mm)")  + 
                                        #scale_linetype_manual(values = c("500 μatm" = "solid", "800 μatm" = "solid", "1200 uatm" = "solid"))  +
                                        theme(legend.position="none") +
                                        theme_classic() +
                                        ggtitle("F1 length") +
                                        # facet_wrap(~factor(Fed_Unfed, level = c('Low food', 'High food'))) 
                                        facet_wrap(~Life_stage, scales = 'free')
F1_length_lineplot_LARVAE


```
















### Grow Out (F1s)

* At age 18 days, bay scallops were moved to the flow through system on downwellers with 200 µm mesh (Fig. 5). OA conditions were maintained in the flow through system by bubbling CO2 into columns prior to entering the exposure tanks and directly into the exposure tanks. 

* We kept initial stocking densities high to assure we had enough for future spawning, but culled regularly to keep biomass from outgrowing the downweller. 

* There were only enough spat produced in the **moderate and low OA treatments** to run the full replicated experiment. There were a total of 16 spat from the high OA treatment that were maintained in the system to test our ability to maintain all three treatments and collected some pilot data to inform experiments with the F2 juveniles. 



<a id="Img_2"></a>
```{r Image 2, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="**Image 2: Experimental flow through set up** Experimental set up with downwellers and columns controlling pCO2 under flow through conditions. Although this does not directly represent the actual pH, we refer to the treatments as pH = 7, 7.5, and 8 for simplicity.", out.width = '80%'}
knitr::include_graphics("C:/Users/samjg/Documents/Github_repositories/Airradians_multigen_OA/Rmd_summary/images/Flow_through_setup.jpg")
```

 

* **Temperature varied overtime**, but was controlled to keep from getting too warm in the summer or too cold in the winter (Fig. 6).  

* **In spring 2022**, F1 bay scallops were held at 16°C to condition for spawning and prevent uncontrolled spawning in the experimental system. While we had target pCO2 concentrations, there was some seasonal variation (Fig. 7) due to extreme variation in the incoming harbor water and therefore is a caveat of long term flow through experimental conditions.  

Figure 6: Changes in temperature overtime for the F1 grow out. 


Figure 7: pCO2 over time during F1 growout












### F1 Physiology 

* Survival counts and shell length measurement were completed monthly and large batches of bay scallops were opportunistically frozen for dry weight measurements as cullings were required to prevent overcrowding. 

* **Respiration Rate and Clearance Rate** were measured approximately monthly during the **early spat stages**. 

* When the bay scallops were **large enough to move into the large respiration chambers** (out of the microrespiration system, 1.7mL), **Excretion Rates were also measured.** 

* Once bay scallops were large enough **(> 20mm) the biodeposition method was used** to measure feeding metrics and clearance Rate.  

* Physiological parameters were **not measured during the overwintering period** when metabolic activity is expected to be low, but resumed during the temperature ramp up in the late winter / early spring. 


**Table 3 Log of physiology time points for F1 bay scallop grow out. **  We are currently using length and weight data collected over the lifespan (September 2021-September 2022) to determine the best method for standardizing physiological rates to accomplish the most appropriate approach to metabolic scaling for our data. Once this is complete, we will include figures for these measurements that will be used to guide data collection during the F2 grow out. 

|Date|Age (days)|Temp|Platform for Respiration|Excretion Rate|Feeding|
|---|---|---|---|---| ---|
|Sept. 14, 2021|50|21°C|1.7mL Loligo chambers (8 channels)|no|Clearance rate (static 20mL vials)|
|Sept. 30 2021|66|20°C|Microrespiration 1.7mL SDR chambers 48 wells| no|Clearance rate (static 20mL vials)|
|Oct. 26, 2021|92|20°C|~23 mL static (8 chambers) |YES|Clearance rate (static 20mL vials)|
|Feb. 2, 2022|162|14°C|~68 mL static (8 chambers) |YES|Clearance rate|
|Mar. 1, 2022|218|16°C|~68 mL static(8 chambers)|YES|Biodeposition|
|Sept. 22, 2022|423|20°C|small closed recirculating (~200 mL)  small recirculating|YES|Biodeposition|
|Oct. 26, 2022|457|12°C|small closed recirculating (~250 mL) large recirculating|YES|Biodeposition|







<a id="Fig_4"></a>
```{r Spat and Adult length plot Mean SE, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="**Figure 8: F1 shell length spat - adult (high OA not included).** Mean +/- se for all data (note: not averaged by tank, but by treatment as whole). ‘Spat’ for all post-metamorphosis stage during initial move to downwellers and < 5mm shell length, ‘juvenile’ for all data >5 mm shell length and before ~8 months in age, ‘adult’ all time points thereafter.", out.width = '100%'}

# Spat to adult length  ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
# read csv file 
F1_length <- read.csv(file= "C:/Users/samjg/Documents/Github_repositories/Airradians_multigen_OA/RAnalysis/Data/Physiology/Length/F1/cumulative_raw/F1_lengths_raw.csv", header = TRUE)
# means for plotting
F1_length_MEANS         <- summarySE(F1_length, measurevar="Length_um", groupvars=c("Age_DPF", "pH", "Life_stage"))
F1_length_MEANS_by_tank <- (summarySE(F1_length, measurevar="Length_um", groupvars=c("Age_DPF", "pH", "Replicate")))

#plot
F1_length_MEANS <- F1_length_MEANS %>% 
  dplyr::mutate(pCO2 = case_when(
    pH == 8 ~ "500 μatm",  
    pH == 7.5 ~ "800 μatm", 
    pH == 7 ~ "1200 μatm"))
F1_length_MEANS$pCO2 <- factor(F1_length_MEANS$pCO2, c("500 μatm","800 μatm","1200 μatm"))

F1_length_lineplot_SPAT.ADULT <- F1_length_MEANS %>% 
                                    dplyr::filter(!Life_stage %in% 'larvae') %>% 
                                    mutate(Life_stage = factor(Life_stage, 
                                                               levels = c("larvae", 
                                                                          "spat", 
                                                                          "juvenile", 
                                                                          "adult"))) %>% # relevel for the facets to occur in correct order
                                    dplyr::filter(!pH %in% 7) %>% 
                                      ggplot(aes(x=Age_DPF, 
                                                 y=(Length_um/1000), 
                                                 color=pCO2)) +
                                        geom_line(size = 0.5) +  # aes(linetype = factor(pCO2)),
                                        geom_point()+ 
                                        scale_color_manual(values=c("forestgreen", "darkorange2"))+
                                        geom_errorbar(aes(ymin=(Length_um/1000)-(se/1000),
                                                          ymax=(Length_um/1000)+(se/1000)), width=.2,
                                                      position=position_dodge(.1))+
                                        theme(panel.grid.major = element_blank(), 
                                              panel.grid.minor = element_blank())+ 
                                        scale_x_continuous(name ="Age (days)") +
                                        scale_y_continuous(name ="Shell length (mm)")  + 
                                        #scale_linetype_manual(values = c("500 μatm" = "solid", "800 μatm" = "solid"))  +
                                        theme(legend.position="none") +
                                        theme_classic() +
                                        ggtitle("F1 length") +
                                        # facet_wrap(~factor(Fed_Unfed, level = c('Low food', 'High food'))) 
                                        facet_wrap(~Life_stage, scales = 'free')

print(F1_length_lineplot_SPAT.ADULT)
```



### Respiration rates 

* rates of oxygen consumption were calculated using LoLinR package following authors' suggested parameters 

  - in all cases, each diagnostic plot was assessed for all rates to ensure the rate with the highest prevalence was represented in the dataset. Noise or incorrect data can occasionally interfere, in these cases the single datapoint was subset or alternative parameters called that better approximate the most prevalent rate

* all rates were normalized/corrected with the following: 

  - corrected for blank rates (mean of blanks within plate/replicate when applicable)
  
  - corrected to volume as (known vessel volume) - (biovolume). Biovolume was measured in most cases, however we used a calculated biovolume  from regression of length^3. This permits an accurate biovolume estimate to back-calculate values for individuals that were only measured for shell length (shell lengths measured in ALL cases)
  
  - bfactor normalized to shell length **<ADD MORE HERE REGARDLING THIS METHOD - BYRNE PAPER>**
  
  
##### Respiration chamber volumes 

We use **two** instruments to measure oxygen conumption rates 

(1) **SDR Sensor plate**: a 24-well plate system for measuring oxygen consumption 
from embryo to spat stage - we have the plates to accomodate up to a well size of 1700ul and tethering up to three plates (72 total reads!). O2 is measured from detection of flourescent dyed-stickers adhered beneath each well. We measure as a static system in water-tight continaers submerged in a water bath - however there exists routes to agitate dishes while controlling temperature (i.e. shaker in an incubator for example)  

(2) **LoLigo 8-channel system**: when larvae/spat outgrow the SDR sensor dish, we move to a dipping probe system. Water is agitated with either magnetic stir bars (in small vessel system) or via an enclosed pump system (for large vessels). We have several sizes to accomodate species development spat - adult with unique vessel volumes listed below:

| Chamber_ID |	Chamber_type |	Chamber_volume |	Instrument |
|--- |--- | ---|--- |
|1|	x_small|	23.1|	loligo_8channel|
|2|	x_small|	23.05	|loligo_8channel|
|3|	x_small	|23.31|	loligo_8channel|
|4|	x_small	|22.55	|loligo_8channel|
|5|	x_small	|23.18	|loligo_8channel|
|6|	x_small	|23.24	|loligo_8channel|
|7|	x_small	|22.63|	loligo_8channel|
|8|	x_small	|22.95|	loligo_8channel|
|--- |--- | ---|--- |
|1|	small	|68.55|	loligo_8channel|
|2|	small	|68.856|	loligo_8channel|
|3|	small	|68.644|	loligo_8channel|
|4|	small	|68.955|	loligo_8channel|
|5|	small	|68.573|	loligo_8channel|
|6|	small	|68.019|	loligo_8channel|
|7|	small	|68.546|	loligo_8channel|
|8|	small	|68.533|	loligo_8channel|
|--- |--- | ---|--- |
|1|	medium	|196.6|	loligo_8channel|
|2|	medium	|200.6|	loligo_8channel|
|3|	medium	|201.4|	loligo_8channel|
|4|	medium	|200|	loligo_8channel|
|5|	medium	|199.2|	loligo_8channel|
|6|	medium|	200.4|	loligo_8channel|
|7|	medium|	201.3|	loligo_8channel|
|8|	medium|	201.7|	loligo_8channel|
|--- |--- | ---|--- |
|1|	large	|254.7	|loligo_8channel|
|2|	large	|257.88|	loligo_8channel|
|3|	large	|246.63|	loligo_8channel|
|4|	large	|252.35|	loligo_8channel|
|5|	large	|242.31|	loligo_8channel|
|6|	large	|247.07|	loligo_8channel|
|7|	large	|241.11|	loligo_8channel|
|8|	large	|243.59|	loligo_8channel|

  




#### Respiration: Biovolume calculated from shell length 

* verify linear relationship between measured biovolume and shell length^3

* obtain a 'calculated' vessel volume 

```{r, biovol vesselvol, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="**Figure XX: biovolume calculated**. In cases when only shell lnegth was measured, it was required to calculate a biovolume representative of those individuals and correct for the volume of the chamber(s). The strong linear relationship supported using the calculated biovolume for all data for all time points estimated with the same standard", out.width = '100%'}
# SET WORKING DIRECTORY :::::::::::::::::::::::::::::::::::::::::::::::
setwd("C:/Users/samjg/Documents/Github_repositories/Airradians_multigen_OA/RAnalysis") # personal computer
# setwd("C:/Users/samjg/Documents/Github_repositories/Airradians_multigen_OA/RAnalysis") # Work computer
# LOAD DATA :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
RR_formatted_F1s <- read.csv(file="C:/Users/samjg/Documents/Github_repositories/Airradians_multigen_OA/RAnalysis/Output/Respiration/RR_F1s_calc_master.csv")

BiovolMEAS_vs_BiovolCALC <- RR_formatted_F1s %>% 
  ggplot(aes(x = Biovolume_g_in_sw, 
             y = Biovol_length3)) + 
  geom_point()+ 
  geom_smooth() + 
  theme_classic() + 
  ggtitle("Biovolume: measured VS. calculated")

VesselvolMEAS_vs_VesselvolCALC <- RR_formatted_F1s %>% 
  ggplot(aes(x = measured_volume, 
             y = calculated_volume)) + 
  geom_point()+ 
  geom_smooth() + 
  theme_classic() + 
  ggtitle("Vessel volume: measured VS. calculated")

ggarrange(BiovolMEAS_vs_BiovolCALC,VesselvolMEAS_vs_VesselvolCALC, nrow = 1, ncol = 2)

```


#### Respiration: Raw(ish) rates

* corrected for blank rates and the calculated vessel volume

```{r, raw umol hr, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="**Figure XX: F1 raw respiration data (umol O2 hr)**. Data is displayed parsed by  developmental stages to facilitate viewing these data.", out.width = '100%'}
# SET WORKING DIRECTORY :::::::::::::::::::::::::::::::::::::::::::::::
setwd("C:/Users/samjg/Documents/Github_repositories/Airradians_multigen_OA/RAnalysis") # personal computer
# setwd("C:/Users/samjg/Documents/Github_repositories/Airradians_multigen_OA/RAnalysis") # Work computer
# LOAD DATA :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
RR_formatted_F1s <- read.csv(file="C:/Users/samjg/Documents/Github_repositories/Airradians_multigen_OA/RAnalysis/Output/Respiration/RR_F1s_calc_master.csv")

F1s.BOXPLOT <- RR_formatted_F1s %>% 
  ggplot(aes(x = factor(Date), 
             y = resp_umol_hr_biovolcalc, #resp_umol_hr, 
             fill = pCO2)) +
         geom_boxplot(alpha = 0.5, # color hue
             width=0.6, # boxplot width
             outlier.size=0, # make outliers small
             position = position_dodge(preserve = "single")) + 
         geom_point(pch = 19, 
                    position = position_jitterdodge(0.01), 
                    size=1) +
         scale_fill_manual(values=c("forestgreen","orange")) +
         theme_classic() + 
         #ylim(0, 0.2) +
         ggtitle("Raw resp") +
         theme(legend.position="none",
               axis.title.y=element_text(size=7),
               axis.title.x=element_text(size=7),
               axis.text.x=element_text(size=7)) +
         stat_summary(fun.y=mean, 
                      geom = "errorbar", 
                      aes(ymax = ..y.., ymin = ..y..), 
                      width = 0.6, 
                      size=0.4, 
                      linetype = "dashed", 
                      position = position_dodge(preserve = "single"))

#F1s.BOXPLOT

F1s.BOXPLOT_facetted <- RR_formatted_F1s %>% 
  ggplot(aes(x = factor(Date), 
             y = resp_umol_hr_biovolcalc, #resp_umol_hr, 
             fill = pCO2)) +
         geom_boxplot(alpha = 0.5, # color hue
             width=0.6, # boxplot width
             outlier.size=0, # make outliers small
             position = position_dodge(preserve = "single")) + 
         geom_point(pch = 19, 
                    position = position_jitterdodge(0.01), 
                    size=1) +
         scale_fill_manual(values=c("forestgreen","orange")) +
         theme_classic() + 
         #ylim(0, 0.2) +
         ggtitle("Raw resp") +
         theme(legend.position="none",
               axis.title.y=element_text(size=7),
               axis.title.x=element_text(size=7),
               axis.text.x=element_text(size=7)) +
         stat_summary(fun.y=mean, 
                      geom = "errorbar", 
                      aes(ymax = ..y.., ymin = ..y..), 
                      width = 0.6, 
                      size=0.4, 
                      linetype = "dashed", 
                      position = position_dodge(preserve = "single")) +
         facet_wrap(~Gen_lifestage, scales = "free")

#F1s.BOXPLOT_facetted

ggarrange(F1s.BOXPLOT_facetted,F1s.BOXPLOT, nrow = 2, ncol = 1)

```

#### Metabolic scaling

* determine the b factor the normalize all data.

* **NOTE**: since we have shell lengths for all individuals (as opposed to tissue dry weights) our team decided upon a b factor based on MO2 and shell length to normalize our data across all life stages

* b factor for all F1 data (based on shell length!) is **2.18**

```{r, b factor shell length, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="**Figure XX: F1 b factor plots** computed from the regression of log(shell length) and log(MO2).", out.width = '100%'}
# SET WORKING DIRECTORY :::::::::::::::::::::::::::::::::::::::::::::::
setwd("C:/Users/samjg/Documents/Github_repositories/Airradians_multigen_OA/RAnalysis") # personal computer
# setwd("C:/Users/samjg/Documents/Github_repositories/Airradians_multigen_OA/RAnalysis") # Work computer
# LOAD DATA :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

# LOAD DATA :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
RR_master <- read.csv(file="Output/Respiration/RR_calc_raw.csv", header=T) %>% 
              filter(!Food %in% 'unfed') %>% # omit low food trial data
              # filter out the F2 measurements
              filter(!Date %in% c('8/30/2022', '11/16/2022')) %>% # an F2 measurement
              filter(!(Date == '9/22/2022' & filetype =='SDR_data')) %>% 
  # unique(RR_master$Date) # "10/26/2021" "2/2/2022"   "3/1/2022"   "8/30/2022"  "9/14/2021"  "9/22/2022"  "9/30/2021"
              dplyr::mutate(Age = case_when(Date == '9/14/2021'  ~ 50,
                                            Date == '9/30/2021'  ~ 66,
                                            Date == '10/26/2021'  ~ 92, 
                                            Date == '2/2/2022' ~ 191,
                                            Date == '3/1/2022'  ~ 218,
                                            Date == '9/22/2022'  ~ 423,
                                            Date == '10/26/2022' ~ 457)) %>% 
              dplyr::mutate(Age = as.factor(Age)) %>% 
              dplyr::arrange(Age)

# ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
# GET B FACTOR FOR ALL AVAILABLE INDIVIDUALS WITH LENGTH AND MO2 ::::::::::::::::::::::::::::::
RR_master_OM               <- RR_master %>% filter(!is.na(resp_umol_hr_biovolcalc)) %>%  filter(!is.na(Length_um)) 
#nrow(RR_master_OM) # 133 (as opposed to only 77 in the previous cunks ommitting TDW NAs!!!!!!!!!!)
RR_master_OM$log10_VO2     <- log10(as.numeric(RR_master_OM$resp_umol_hr_biovolcalc)) # assign resp value
RR_master_OM$log10_Length  <- log10(as.numeric(RR_master_OM$Length_um)) # assign length value 
#summary(lm(RR_master_OM$log10_VO2~RR_master_OM$log10_Length)) # 1.8502 == b factor

MO2_b.factorLENGTH_PLOT <- RR_master_OM %>% 
                           #filter(!resp_umol_L_hr < 0.08) %>% 
                           ggplot(aes(x=log10_Length, y=log10_VO2)) +
                              geom_point() +
                              theme(panel.grid.major = element_blank(), 
                                    panel.grid.minor = element_blank())+ 
                              scale_x_continuous(name ="log10_Length; in mm") +
                              scale_y_continuous(name ="log10_MO2; RR in umol L-1 hr-1)") +
                              theme_classic() +
                              theme(legend.position="none",axis.title.y=element_text(size=7)) +
                              ggtitle("Metabolic scaling: log10_MO2 = log10_a + 
                                      (b.factor * log10_Length)") +
                              geom_smooth(method = lm, color = 'red') +
                              ggpmisc::stat_poly_eq(parse=T, 
                                                    aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~")),
                                                    label.x.npc = "left")
#MO2_b.factorLENGTH_PLOT



MO2_b.factorLENGTH_PLOT_facetted <- RR_master_OM %>% 
                                     ggplot(aes(x=log10_Length, y=log10_VO2, color = pCO2)) +
                                        geom_point() +
                                        scale_color_manual(values=c("blue","orange")) +
                                        theme(panel.grid.major = element_blank(), 
                                              panel.grid.minor = element_blank())+ 
                                        scale_x_continuous(name ="log10_Length; in mm") +
                                        scale_y_continuous(name ="log10_MO2; RR in umol L-1 hr-1)") +
                                        theme_classic() +
                                        theme(legend.position="none",axis.title.y=element_text(size=7)) +
                                        ggtitle("Metabolic scaling: log10_MO2 = log10_a + 
                                                (b.factor * log10_Length)") +
                                        geom_smooth(method = lm, color = 'red') +
                                        ggpmisc::stat_poly_eq(parse=T, 
                                                              aes(label = paste(..eq.label..,
                                                                                ..rr.label.., sep = "~~~")),
                                                              label.x.npc = "left") +
                                        facet_wrap(~pCO2)
# MO2_b.factorLENGTH_PLOT_facetted


# OUTPUT PLOTS 
print(ggarrange(MO2_b.factorLENGTH_PLOT, MO2_b.factorLENGTH_PLOT_facetted, nrow = 2, ncol = 1)) # print the model diagnostics
```


#### Shell length normalized 

```{r Length Normalized Respiration Rate, echo=FALSE, message=FALSE, warning=FALSE, fig.cap="**Figure XX: F1 Respiration rate (shell length normalized)** Mean +/- se for all data employing the b factor 2.18 (as shown above).", out.width = '100%'}
# SET WORKING DIRECTORY :::::::::::::::::::::::::::::::::::::::::::::::
setwd("C:/Users/samjg/Documents/Github_repositories/Airradians_multigen_OA/RAnalysis") # personal 

# LOAD DATA :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
RR_formatted_F1s <- read.csv(file="C:/Users/samjg/Documents/Github_repositories/Airradians_multigen_OA/RAnalysis/Output/Respiration/RR_F1s_calc_master.csv")

F1s.BOXPLOT_Length_bfactor.MEAN <- RR_formatted_F1s %>% 
  ggplot(aes(x = factor(Date), 
             y = resp_umol_hr_bFactorNormLength.MEAN, 
             fill = pCO2)) +
         geom_boxplot(alpha = 0.5, # color hue
             width=0.6, # boxplot width
             outlier.size=0, # make outliers small
             position = position_dodge(preserve = "single")) + 
         geom_point(pch = 19, 
                    position = position_jitterdodge(0.01), 
                    size=1) +
         scale_fill_manual(values=c("forestgreen","orange")) +
         theme_classic() + 
         ggtitle("Resp * [ ( meanLength / Length_meas)^2.18 ]") +
         theme(legend.position="none",
               axis.title.y=element_text(size=7),
               axis.title.x=element_text(size=7),
               axis.text.x=element_text(size=7)) +
         #ylim(0, 0.2) +
         stat_summary(fun.y=mean, 
                      geom = "errorbar", 
                      aes(ymax = ..y.., ymin = ..y..), 
                      width = 0.6, 
                      size=0.4, 
                      linetype = "dashed", 
                      position = position_dodge(preserve = "single")) +
         facet_wrap(~Gen_lifestage, scales = "free") #, scales = "free)

F1s.BOXPLOT_Length_bfactor.MEAN.nofacet <- RR_formatted_F1s %>% 
  ggplot(aes(x = factor(Date), 
             y = resp_umol_hr_bFactorNormLength.MEAN, 
             fill = pCO2)) +
         geom_boxplot(alpha = 0.5, # color hue
             width=0.6, # boxplot width
             outlier.size=0, # make outliers small
             position = position_dodge(preserve = "single")) + 
         geom_point(pch = 19, 
                    position = position_jitterdodge(0.01), 
                    size=1) +
         scale_fill_manual(values=c("forestgreen","orange")) +
         theme_classic() + 
         ggtitle("Resp * [ (meanLength/Length_meas)^2.18 ]") +
         theme(legend.position="none",
               axis.title.y=element_text(size=7),
               axis.title.x=element_text(size=7),
               axis.text.x=element_text(size=7)) +
         #ylim(0, 0.2) +
         stat_summary(fun.y=mean, 
                      geom = "errorbar", 
                      aes(ymax = ..y.., ymin = ..y..), 
                      width = 0.6, 
                      size=0.4, 
                      linetype = "dashed", 
                      position = position_dodge(preserve = "single")) # no facet

# save
print(ggarrange(F1s.BOXPLOT_Length_bfactor.MEAN,F1s.BOXPLOT_Length_bfactor.MEAN.nofacet, nrow = 2, ncol = 1)) # print the model diagnostics 

```


### Clearance rates 

* units as cells per mL per hour 

* parsed by cell species and type as Chaetocerous, Tetraselmis (Ply), and seston

* length b factor normalized **2.18** 


```{r clearance rates}
# SET WORKING DIRECTORY :::::::::::::::::::::::::::::::::::::::::::::::
setwd("C:/Users/samjg/Documents/Github_repositories/Airradians_multigen_OA/RAnalysis") # personal computer
# setwd("C:/Users/samjg/Documents/Github_repositories/Airradians_multigen_OA/RAnalysis") # Work computer
# LOAD DATA :::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
CR_master <- read.csv(file="Output/ClearanceRates/Calculated_Clearance_Master.csv", header=T)

Master_914  <- CR_master  %>% dplyr::filter(Date %in% '20210914')
Master_930  <- CR_master  %>% dplyr::filter(Date %in% '20210930')
Master_1026 <- CR_master  %>% dplyr::filter(Date %in% '20211026')
Master_0202 <- CR_master  %>% dplyr::filter(Date %in% '20220202')

CR_914_boxplot <- Master_914 %>% 
                              ggplot(aes(pCO2 , 
                                         #CR_mL.hr.mmlength , 
                                         #CR_mL.hr.length,
                                         CR_mL.hr.bfactorlength,
                                         fill = pCO2)) +
                              theme(panel.grid=element_blank()) +
                              geom_boxplot(size=0.2,
                                           alpha=0.1, 
                                           aes(fill=pCO2)) +
                              scale_fill_manual(values=c("white", 
                                                         "grey50")) +
                              geom_point(shape = 21, 
                                         size = 2, 
                                         position = position_jitterdodge(jitter.width = 0.1)) +
                              theme_classic() +
                              theme(axis.text=element_text(size=12),
                                    axis.title=element_text(size=10)) +
                              stat_summary(fun.y=mean, 
                                           geom="point", 
                                           shape=18, 
                                           size=4, 
                                           color="black", 
                                           fill="white") +
                              ggtitle("Clearance rate, F1 Scallops 20210914") +
                              facet_wrap(~type)

CR_930_boxplot <- Master_930 %>% 
                              ggplot(aes(pCO2 , 
                                         #CR_mL.hr.mmlength , 
                                         #CR_mL.hr.length,
                                         CR_mL.hr.bfactorlength,
                                         fill = pCO2)) +
                              theme(panel.grid=element_blank()) +
                              geom_boxplot(size=0.2,
                                           alpha=0.1, 
                                           aes(fill=pCO2)) +
                              scale_fill_manual(values=c("white", 
                                                         "grey50")) +
                              geom_point(shape = 21, 
                                         size = 2, 
                                         position = position_jitterdodge(jitter.width = 0.1)) +
                              theme_classic() +
                              theme(axis.text=element_text(size=12),
                                    axis.title=element_text(size=10)) +
                              stat_summary(fun.y=mean, 
                                           geom="point", 
                                           shape=18, 
                                           size=4, 
                                           color="black", 
                                           fill="white") +
                              ggtitle("Clearance rate, F1 Scallops 20210930") +
                              facet_wrap(~type)


CR_1026_boxplot <- Master_1026 %>% 
                              ggplot(aes(pCO2 , 
                                         #CR_mL.hr.mmlength , 
                                         #CR_mL.hr.length,
                                         CR_mL.hr.bfactorlength,
                                         fill = pCO2)) +
                              theme(panel.grid=element_blank()) +
                              geom_boxplot(size=0.2,
                                           alpha=0.1, 
                                           aes(fill=pCO2)) +
                              scale_fill_manual(values=c("white", 
                                                         "grey50")) +
                              geom_point(shape = 21, 
                                         size = 2, 
                                         position = position_jitterdodge(jitter.width = 0.1)) +
                              theme_classic() +
                              theme(axis.text=element_text(size=12),
                                    axis.title=element_text(size=10)) +
                              stat_summary(fun.y=mean, 
                                           geom="point", 
                                           shape=18, 
                                           size=4, 
                                           color="black", 
                                           fill="white") +
                              ggtitle("Clearance rate, F1 Scallops 20211026") +
                              facet_wrap(~type)


CR_0202_boxplot <- Master_0202 %>% 
                              ggplot(aes(pCO2 , 
                                         #CR_mL.hr.mmlength , 
                                         #CR_mL.hr.length,
                                         CR_mL.hr.bfactorlength,
                                         fill = pCO2)) +
                              theme(panel.grid=element_blank()) +
                              geom_boxplot(size=0.2,
                                           alpha=0.1, 
                                           aes(fill=pCO2)) +
                              scale_fill_manual(values=c("white", 
                                                         "grey50")) +
                              geom_point(shape = 21, 
                                         size = 2, 
                                         position = position_jitterdodge(jitter.width = 0.1)) +
                              theme_classic() +
                              theme(axis.text=element_text(size=12),
                                    axis.title=element_text(size=10)) +
                              stat_summary(fun.y=mean, 
                                           geom="point", 
                                           shape=18, 
                                           size=4, 
                                           color="black", 
                                           fill="white") +
                              ggtitle("Clearance rate, F1 Scallops 20220202") +
                              facet_wrap(~type)

print(ggarrange(CR_914_boxplot,CR_930_boxplot,CR_1026_boxplot,CR_0202_boxplot))
```
 Figure 10a:  Clearance rates. Updated as of 20221031. Note, these do not have a b factor correction

Figure 10b:  Clearance rates of juveniles using static methods. 


 Figure 11: Biodeposition metrics for F1s. Updated 20221024. 
Cumulative_data_file -note if changes are made in the google drive worksheet it must translate to this file), calculation script; master data file;  ANOVA tables. Three significant effects (one-way anova; residuals normal via shapiro wilk and levenes test) on 20220923 (temperature 20C) for (1)Cumulative_data_file selection efficiency (2) % rejection rate (3) assimilation efficiency. Temperature in March was 16°C and temperature in August was 20°C
TSM is consistent between the two treatments, but not among dates/temperatures. This means that they have different food availability (quality and quantity) across timepoints 
Make a table (or plot) of the % organic material and TSM across the various timepoints - table for the ‘input’ and ‘blank’ samples (updated as of Nov. 2022)
Note, 20C time point contains some release gametes for a few individiuals




 
Figure 12: Excretion data for F1 scallops - bfactor normalized using MO2 TDW (page 3, b =0833; updated as of 20221214)
Excretion rate data: script; cumulative raw; master file; plots


Figure 13a: Dry shell and dry tissue F1s. Updated 20221118
F1 Dry shell (g, top) and dry tissue weight (g, bottom) as mean SE measured 72 hours at 60C. Right panel depicts data normalized to individual shell lengths. github live link


Figure 13b: Dry shell and dry tissue F2s. Updated 20221118
F2 Dry shell (top) and dry tissue weight (bottom) as mean SE measured 72 hours at 60C. Right panel depicts data normalized to individual shell lengths. github live link
 Gonad development was quantified by destructively sampling for gonad index approximately monthly from late February through June 2022 (Fig. 12). 
 Figure 12: Gonad index over time. 

Figure 13: Percent muscle over time.  

### Gene expression (F1s)

<add here>


### SNPs & Genotype liklihoods (F1s)

<add here>

## F2: First Generation

---



**Table 2: Average Water Chemistry during the F2 grow out**
(last updated 10/18/22). Using DIC and pH for the calculations.

Treatment
pCO2
pH
ΩAr
High OA
1177
7.6
0.8
Mid OA
751
7.8
1.1
Low OA
543
7.9
1.4



### Larval development (F2s)


Several spawning attempts were made. Gonad development was evident in April and the first successful spawn was conducted. While spawning and fertilization was successful in April, after ~5 days of the full factorial experiment, it was evident that the larvae were crashing and after 10 days, 100% mortality was observed. Cause unknown. A second successful spawn was conducted in early July 2022, but at the water change and check at age 2 days old, heavy mortality was observed. After investigation, it was concluded that Vibrio was introduced through the food and the problem was fixed. A third and final spawn was conducted in late August 2022 with success (details below), but the egg output was low and we were unable to initiate the full factorial exposure and instead focused on just the creation of the F2 cohorts for each respective OA treatment (low and moderate - no high OA broodstock). Based on survival rates to date, we believe this was a wise decision. Fertilization rate was high (to first polar body), but progression to the multicellular stage after 4 hours was low (~20%). Food limitation (serious oyster population growing in the pipes from the harbor to the building) and several accidental spawnings due to pump failures while ripe scallops were being held could be part of the cause for a low output during this spawn. 

### Spawning and fertilization

Spawning was induced using thermal adjustments up to 25°C. When an individual was observed to be spawning, it was removed from the spawning table and placed in a 1-L beaker with filtered seawater at 25°C to continue releasing gametes. To avoid self fertilization, scallops were switched to new beakers frequently to isolate sperm and egg from each individual. When eggs were observed to be released, the animal was immediately switched to a new cup and eggs were caught on a 25µm sieve to remove any sperm that remained in the beaker. Prior to fertilization, sperm was rinsed through a 20µm sieve to remove any eggs that may have remained in the cup. Fertilizations were completed in batches to assure gametes were fertilized within approximately 1 hour of release and for each batch fertilization, each individual used contributed only sperm or only eggs to avoid self fertilization. Embryos were checked frequently to assure fertilization was occurring and gametes were not pooled until > 85% of the embryos showed clear cell division. Spawnings for the two cohorts (low and moderate OA) were conducted simultaneously and the number of males and females contributing to each cohort were kept constant. Eggs were used from a total of 11 individuals and sperm from a total of 11 individuals, of these 4 individuals supplied both sperm and egg, but never to the same fertilization set. 

Exposure to treatments
	After approximately 4 hours post fertilization, embryos from all batches were pooled and divided among 2 OA treatments (~400 and 800 µatm) with 3 replicates per treatment. Exposures were conducted in 10L filtered seawater at 27 psu and 23°C. Eggs were overstocked because of observations of stunted fertilization, so an initial embryo density was not accurately determined. Day 2 stocking density was 5-6 larvae / mL. Details of the chemistry for each OA treatment are detailed in Table 4. 



Table 4: Average Water Chemistry during the F2 larval experiment 
Using DIC and pH for the calculations.
Treatment
pCO2
pH
ΩAr
Mid OA
787
7.8
1.5
Low OA
386
8.0
2.6






Table 5: Endpoint data summary for the F2 larval exposure to low and moderate OA
Date is from age 21 days old (last official count before moving to the basement. Percent Successful Metamorphosis =  Live spat / (Dead Spat + Live spat). Plots to come…


Low OA
Moderate OA
Mean Shell Length (µm)
350.6 µm
320.6 µm
Mean Survival (%)
2.70%
0.90%
Mean Successful Metamorphosis (%)
59.4%
8.8%





Figure A: Larval survival during static exposures. 



Figure B: F2 Larvae mean shell length over time



Metamorphosis
Successful metamorphosis was quantified at age 21 days by counting the number of live spat and dead spat in each replicate. In the Moderate OA treatments, there were high rates of dead, but metamorphosed spat (Fig. C).




Figure C: Percent successful metamorphosis (F2)

### Grow Out (F2s)

F2 bay scallops were moved to the basement flow through system at age 23 days post fertilization and reared in downwelllers on 200µm mesh on drumfiltered seawater (30µm) at a temperature of ~22°C and salinity of 27-29. Spat were evenly distributed across 12 downwellers (n=4 for pH = 7.5 and n = 8 for pH =8). The first real counts were conducted 2 weeks after moving to the basement (9/21/22; age 36 dpf) and all downwellers were standardized to ~100 individuals per downweller. In addition to food coming in with the seawater from the harbor, diet was also supplemented with cultured phytoplankton (Chaet, T. iso, and PLY). 

The spat produced from pH=7.5 F1 broodstock reared in pH = 7.5 during the larval stage will continue to be reared in a pH of 7.5 for their lifespan (n=4). The pat produced from pH=8 F1 broodstock reared in pH = 8 during the larval stage, were split between two treatments for grow out, pH = 8 (n=4) and pH = 7 (n=4). The change from pH =8 to pH = 7 was gradual (0.1 pH units per week) and they were given a 2 week acclimation period to the flow through system before the pH drop was initiated. pH was slowly decreased over ~3 weeks
9/8/22 - moved to the basement (counts estimated based on volume, no length measurements)
9/21/22 - 2 wks after the move: pH still same as low OA treatment (length measurements and first real counts)
10/5/22 - 4 wks after the move: pH at to the mid treatment (length measurements and counts)
10/26/22 - 7 wks after the move: pH at to the high treatment (length measurements and counts)
11/16/22 - 10 weeks after the move: period of acclimation to the high OA treatment (Respiration Rates, length measurement, and counts)

The rationale for this is to obtain some physiological data under more extreme OA conditions. Because we were not able to produce larvae that could survive the extreme condition, exposure beginning as spat will help us to understand if other life stages are resilient to OA. Due to low survival in the pH = 7.5 during the static larval exposure, we did not have the numbers needed to also expose pH=7.5 spat to a pH=7 without risk of losing numbers and broodstock for a spawn to create the F3 generation, which is the primary goal of this project. 

Growth and survival will be monitored approximately monthly and physiology will be measured opportunistically when destructive sampling is feasible (ie. the main focus is rearing enough to have good broodstock numbers, so destructive sampling will only occur when a culling is required to reduce densities). 

Destructive sampling for genomics will be conducted during the first large culling requirement (November 17-18, 2022) and only the minimum number required will be sampled.



Figure D: pH over time during the change in pH from 8 to 7 in the low OA → high OA cohort/treatment. 
Upward drift do to the loss in the phytoplankton bloom in the harbor requiring more adjustments to the system 



Figure E: pCO2 over time during the change in pH from 8 to 7 in the low OA → high OA cohort/treatment. 






Figure D: Mean shell growth rate (mm / month) 
Growth rate of spat during their first month in the flow through system. The first time point (September 21, 2022), the High OA treatment was still in its home OA (Low OA; treatment larvae were reared) and by the October 5, 2022 length measurements, the High OA treatment had reach approximately the same OA conditions as the moderate treatment. At the October 26 measurements theHigh OA treatment will have reached the High OA treatment target. 



Figure F: F2 juvenile growth: Mean shell length over time.
The high OA treatment (1200 µatm) started with juveniles raised in the loa OA treatment (500 µatm) and were slowly adjusted. At day 50, they reached the moderate OA treatment and at 71 days old, they reached the high OA treatment. 

Table X: Log of physiology time points for F2 bay scallop growout.  
Date
Age
(dpf)
Temp
Platform for Respiration
Excretion
Feeding
Notes
Aug 30, 2022
14
23°C
Microrespiration
(80uL chambers)
no
no
Pre and post metamorphic (static)
Sept. 21, 2022
36
20°C
Microrespiration 
(80uL chambers)
no
no
Postset 2 wks after moving to flow through
-extra low OA dropped to mod OA
Nov. 16, 2022
92
22°C
60 m static chambers
YES
no


Jan ??


12°C
60 mL static chambers






May or June?


16°C
big closed recirculating 






### Physiology (F2s)



<add here>



### Gene expression (F1s)



<add here>



### SNPs & Genotype liklihoods (F1s)



<add here>



## Chemistry 
---

Discrete chemistry was measured daily for dissolved oxygen (mg L-1), salinity, and temperature (°C) in each treatment tank (YSI Model 556, Yellowspring, OH USA). Seawater samples were collected weekly in dark polypropylene bottles (500 mL) from each tank to measure pH, dissolved organic carbon (DIC, μmol kg-1), and total alkalinity (TA, μmol kg-1) via open-cell titration (SOP 3b; Dickson et al., 2007). DIC was measured on an Apollo SciTech DIC analyzer (Apollo SciTech, LLC, Newark, DE) which had a precision of 0.5% of assigned values in an interlaboratory comparison (Bockmon & Dickson, 2015).  TA was measured using certified HCl titrant (∼0.1 mol kg−1, ∼0.6 mol kg−1 NaCl; Dickson Lab, Batches 191 and 157) on a Metrohm alkalinity titrator (Mettler Toledo add here) with 0.17% mean error relative to certified reference materials (Dickson Lab CO2 CRM Batch 191 and 157). A UV-VIS spectrophotometer (Cary100, Agilent, Santa Clara, CA, USA) was used to determine pH colorimetrically at 20oC with m-cresol purple indicator dye (Sigma-Aldrich, St. Louise, MS, Dickson & Goyet, 1994), with an error of ±0.0014 (seawater scale). Seawater DIC and pH were used in CO2SYS (Pierrot et al., 2006) for the calculation of partial pressure pCO2 (μatm), carbon ion constituents (bicarbonate HCO3- and carbonate CO3-2, μmol kg-1), and aragonite and calcite saturation state (Ωar and Ωca, respectively) using the following constants: K1, K2 from Lueker et al. (2000); potassium sulfate from Dickson (1990); and boron from Lee et al. (2010).



Figure C1: Changes in water chemistry over time during F1 growout




Figure C2: Changes in water chemistry over time during F1 growout



Food 
CHGRA - Chaetoceros muelleri
Chaet B - Chaetoceros neogracile
Ply429 - Tetraselmis chui
Mono - Diacronema lutheri (previously Pavlova lutheri)



<script type="text/javascript">
  // When the document is fully rendered...
  $(document).ready(function() {
    // ...select all header elements...
    $('h1, h2, h3, h4, h5').each(function() {
      // ...and add an id to them corresponding to their 'titles'
      $(this).attr('id', $(this).html());
    });
  });
</script>


<script type="text/javascript">
  $(document).ready(function() {
    $('pre.r').each(function(i) {
      $(this).attr('id', 'Chunk_' + i);
    });
  });
</script>


